Codec/Archive/Tar/Types.hs

```diff
@@ -563,13 +564,20 @@ instance Arbitrary Entry where
                     <*> arbitrary <*> arbitraryEpochTime <*> arbitrary
     where
       arbitraryPermissions :: Gen Permissions
-      arbitraryPermissions = fromIntegral <$> (arbitraryOctal 7 :: Gen Int)
+      arbitraryPermissions = fromIntegral <$> permGen
+        where fmodeGen n   = (flip shiftL) n <$> choose (0, 7) :: Gen Int
```

Comment:
would you mind explaining the rationale about this? I have to admit I don't see right away what configuration space this expression is trying to cover... the  change below had the helpful

```
-- restrict user/group to posix ^[a-z][-a-z0-9]{0,30}$
```

comment which was easy to correlate to the code. Is this possible here as well?

Reply:
Sure. `arbitraryOctal 7` doesn't work for Permissions, because it overflows the underlying `Permissions` type. `Permissions` is a type alias for `FileMode` which in turn is an alias for the `CMode` newtype from `System.Posix.Types`. CMode is corresponds to the posix`mode_t` from `chmod`, which is 16bit unsigned int.
`8^7-1 == 2097151` `2^16 == 65536`

I could simplify this by changing `arbitraryPermissions` to be just 
```
arbitraryPermissions = fromIntegral <$> (arbitrary :: Gen Word16)
```
(which might be better than individually generating the `chmod` mode bits and combining to a `FileMode` like I originally did in the PR)